program/src/accounts.rs

```diff
@@ -340,9 +340,7 @@ macro_rules! accounts_struct {
 
                     // Also consume the iterator, so the no-excess-accounts check
                     // below does not trigger.
-                    for _ in 0..$multi_account.len() {
-                        accounts_iter.next();
-                    }
+		    let mut accounts_iter = accounts_iter.skip($multi_account.len());
```

Comment:
You are re-defining `accounts_iter` here ... does it actually remain in scope (or rather, escape the `$(...)?` scope), such that the next `accounts_iter` reference below references the skipped version, and not the original one?

I found the former version to be easier to understand, because `.skip` does not actually mutate the existing iterator to skip that many elements, it wraps the iterator in a _new_ one.

Reply:
It shadows previous definition of `accounts_iter`. 
`std::slice::Skip` also implements same Iterator trait as `std::slice::Iter`, so no difference and it's easier to read. But if you insist, I will revert it back